### ISSUE TITLE
Getting TypeError when I change use_cudnn_rnn to false.

### ISSUE THREAD

**Issue by mnsrmov** [deepmipt/DeepPavlov#527]
Whenever I change the use_cudnn_rnn value from true to false in the ner_ontonotes.json file I get this error:

> TypeError: _build_rnn() takes 5 positional arguments but 6 were given

This is the full body of the error:

```
2018-10-15 13:31:02.902 ERROR in 'deeppavlov.core.common.params'['params'] at line 110: Exception in <class 'deeppavlov.models.ner.network.NerNetwork'>
Traceback (most recent call last):
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\common\params.py", line 104, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\models\ner\network.py", line 151, in __init__
    units = self._build_rnn(features, n_hidden_list, cell_type, intra_layer_dropout, self.mask_ph,)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
TypeError: _build_rnn() takes 5 positional arguments but 6 were given
Traceback (most recent call last):
  File "D:/PythonWorkSpace/DeepPavlovAttempt_4/test2.py", line 9, in <module>
    train_evaluate_model_from_config(PIPELINE_CONFIG_PATH)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\commands\train.py", line 174, in train_evaluate_model_from_config
    model = fit_chainer(config, iterator)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\commands\train.py", line 69, in fit_chainer
    component = from_params(component_config, mode='train')
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\common\params.py", line 104, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\models\ner\network.py", line 151, in __init__
    units = self._build_rnn(features, n_hidden_list, cell_type, intra_layer_dropout, self.mask_ph,)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
TypeError: _build_rnn() takes 5 positional arguments but 6 were given 
```

Any help is highly appreciated. And just in case, I didn't paste it twice. That is how it appears.

**Comment by mu-arkhipov** [comment]
This issue will be fixed next release. It will be in days to come

**Comment by mu-arkhipov** [comment]
At the monent you can use dev branch and set use use_cudnn to true, the CPU version of RNN wil be used automatically.

**Comment by mnsrmov** [comment]
Thank you very much for the response @mu-arkhipov ! However, now I'm getting the error below. I am using the dev branch this time. That is one error.
```
2018-10-16 11:51:42.509 ERROR in 'deeppavlov.core.common.params'['params'] at line 109: Exception in <class 'deeppavlov.models.ner.network.NerNetwork'>
Traceback (most recent call last):
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1322, in _do_call
    return fn(*args)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1307, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1409, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\common\params.py", line 103, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\models\ner\network.py", line 169, in __init__
    self.load()
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_model.py", line 47, in load
    saver.restore(self.sess, path)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1752, in restore
    {self.saver_def.filename_tensor_name: save_path})
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 900, in run
    run_metadata_ptr)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1135, in _run
    feed_dict_tensor, options, run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1316, in _do_run
    run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1335, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]

Caused by op 'save/Assign_17', defined at:
  File "D:/PythonWorkSpace/DeepPavlov_Dev/test1.py", line 10, in <module>
    train_evaluate_model_from_config(PIPELINE_CONFIG_PATH)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 200, in train_evaluate_model_from_config
    model = fit_chainer(config, iterator)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 87, in fit_chainer
    component = from_params(component_config, mode='train')
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\common\params.py", line 103, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\models\ner\network.py", line 169, in __init__
    self.load()
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_model.py", line 46, in load
    saver = tf.train.Saver(var_list)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1284, in __init__
    self.build()
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1296, in build
    self._build(self._filename, build_save=True, build_restore=True)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1333, in _build
    build_save=build_save, build_restore=build_restore)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 781, in _build_internal
    restore_sequentially, reshape)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 422, in _AddRestoreOps
    assign_ops.append(saveable.restore(saveable_tensors, shapes))
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 113, in restore
    self.op.get_shape().is_fully_defined())
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\state_ops.py", line 219, in assign
    validate_shape=validate_shape)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\gen_state_ops.py", line 59, in assign
    use_locking=use_locking, name=name)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\ops.py", line 3414, in create_op
    op_def=op_def)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\ops.py", line 1740, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]

Traceback (most recent call last):
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1322, in _do_call
    return fn(*args)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1307, in _run_fn
    options, feed_dict, fetch_list, target_list, run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1409, in _call_tf_sessionrun
    run_metadata)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:/PythonWorkSpace/DeepPavlov_Dev/test1.py", line 10, in <module>
    train_evaluate_model_from_config(PIPELINE_CONFIG_PATH)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 200, in train_evaluate_model_from_config
    model = fit_chainer(config, iterator)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 87, in fit_chainer
    component = from_params(component_config, mode='train')
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\common\params.py", line 103, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\models\ner\network.py", line 169, in __init__
    self.load()
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_model.py", line 47, in load
    saver.restore(self.sess, path)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1752, in restore
    {self.saver_def.filename_tensor_name: save_path})
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 900, in run
    run_metadata_ptr)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1135, in _run
    feed_dict_tensor, options, run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1316, in _do_run
    run_metadata)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\client\session.py", line 1335, in _do_call
    raise type(e)(node_def, op, message)
tensorflow.python.framework.errors_impl.InvalidArgumentError: Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]

Caused by op 'save/Assign_17', defined at:
  File "D:/PythonWorkSpace/DeepPavlov_Dev/test1.py", line 10, in <module>
    train_evaluate_model_from_config(PIPELINE_CONFIG_PATH)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 200, in train_evaluate_model_from_config
    model = fit_chainer(config, iterator)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\commands\train.py", line 87, in fit_chainer
    component = from_params(component_config, mode='train')
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\common\params.py", line 103, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 75, in __call__
    obj.__init__(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\models\ner\network.py", line 169, in __init__
    self.load()
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_backend.py", line 27, in _wrapped
    return func(*args, **kwargs)
  File "D:\LocalDevs\SlotFilling\DeepPavlov-dev\deeppavlov\core\models\tf_model.py", line 46, in load
    saver = tf.train.Saver(var_list)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1284, in __init__
    self.build()
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1296, in build
    self._build(self._filename, build_save=True, build_restore=True)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 1333, in _build
    build_save=build_save, build_restore=build_restore)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 781, in _build_internal
    restore_sequentially, reshape)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 422, in _AddRestoreOps
    assign_ops.append(saveable.restore(saveable_tensors, shapes))
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\training\saver.py", line 113, in restore
    self.op.get_shape().is_fully_defined())
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\state_ops.py", line 219, in assign
    validate_shape=validate_shape)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\ops\gen_state_ops.py", line 59, in assign
    use_locking=use_locking, name=name)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\op_def_library.py", line 787, in _apply_op_helper
    op_def=op_def)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\ops.py", line 3414, in create_op
    op_def=op_def)
  File "D:\Apps\Anaconda3\envs\py36\lib\site-packages\tensorflow\python\framework\ops.py", line 1740, in __init__
    self._traceback = self._graph._extract_stack()  # pylint: disable=protected-access

InvalidArgumentError (see above for traceback): Assign requires shapes of both tensors to match. lhs shape= [9,9] rhs shape= [37,37]
	 [[Node: save/Assign_17 = Assign[T=DT_FLOAT, _class=["loc:@transitions"], use_locking=true, validate_shape=true, _device="/job:localhost/replica:0/task:0/device:CPU:0"](transitions, save/RestoreV2:17)]]
```

**Comment by mnsrmov** [comment]
I found the problem. So the problem was that I had a mistake in the model I was reading and it didn't give me a comprehensive error. It took me a while to find it. Furthermore, I noticed that fasttext_embedder.py imports fastText. This is ok if the package is not installed but imported as an external resource. If the fastText package is installed using pip however, it has to be imported as `fasttext`. No capital letters. Additionally, it may be good to specify which fasttext package you are using. I had to test three different packages until I found one which doesn't fail at line 78 of the said file. This: `Fasttext.FastText._FastText` does not work for all packages. At the end, thanks again for the help and keep up the good work!
